mathics/builtin/numbers/algebra.py

```diff
@@ -1351,11 +1351,11 @@ def apply(self, expr, form, h, evaluation):
             return Expression(SymbolDirectedInfinity, Integer(-1))
 
         if not form.has_form("List", None):
-            return Expression(h, *[i for i in find_exponents(expr, form)])
+            return Expression(h, *[i for i in get_exponents_sorted(expr, form)])
```

Comment:
As mentioned above, in the expression refactor branch we can make use of knowing that coefficients will be ordered.